addon/components/lt-body.js

```diff
@@ -266,17 +267,31 @@ export default Component.extend({
       let isSelected = row.get('selected');
       let currIndex = rows.indexOf(row);
       let prevIndex = this._prevSelectedIndex === -1 ? currIndex : this._prevSelectedIndex;
+      let affectedSelections = Ember.A();
 
       this._currSelectedIndex = currIndex;
       this._prevSelectedIndex = prevIndex;
 
       if (canSelect) {
         if (e.shiftKey && multiSelect) {
-          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => r.set('selected', !isSelected));
+          rows.slice(Math.min(currIndex, prevIndex), Math.max(currIndex, prevIndex) + 1).forEach(r => {
+            if (!r.get('selected')) {
+                affectedSelections.pushObject(r);
+            }
+
+            r.set('selected', !isSelected);
+          });
           this._prevSelectedIndex = currIndex;
         } else if ((!multiSelectRequiresKeyboard || (e.ctrlKey || e.metaKey)) && multiSelect) {
           row.toggleProperty('selected');
+          affectedSelections.pushObject(row);
         } else {
+          if (!isSelected) {
+            affectedSelections.pushObject(row);
```

Comment:
Why is this only pushed if `isSelected` is false? 

No matter what the situation is, if `canSelect` is true, the target row will be affected. I really dont think we need to add it to the list. What if we generalized it a bit more?

What if affectedSelections becomes a collection of other rows that were affected by the click event? This can include both `expanded` and `selected`.